components/eam/src/physics/cam/ndrop.F90

```diff
@@ -306,190 +306,166 @@ subroutine dropmixnuc( &
-   real(r8) :: wbar, wmix, wmin, wmax
-
-   real(r8) :: zn(pver)   ! g/pdel (m2/g) for layer
-   real(r8) :: flxconv    ! convergence of flux into lowest layer
-
-   real(r8) :: wdiab           ! diabatic vertical velocity
-   real(r8) :: ekd(pver)       ! diffusivity for droplets (m2/s)
-   real(r8) :: ekk(0:pver)     ! density*diffusivity for droplets (kg/m3 m2/s)
-   real(r8) :: ekkp(pver)      ! zn*zs*density*diffusivity
-   real(r8) :: ekkm(pver)      ! zn*zs*density*diffusivity
-
-   real(r8) :: dum, dumc
-   real(r8) :: tmpa
-   real(r8) :: dact
-   real(r8) :: fluxntot         ! (#/cm2/s)
-   real(r8) :: dtmix
-   real(r8) :: alogarg
-   real(r8) :: overlapp(pver), overlapm(pver) ! cloud overlap
-
-   real(r8) :: nsource(pcols,pver)            ! droplet number source (#/kg/s)
-   real(r8) :: ndropmix(pcols,pver)           ! droplet number mixing (#/kg/s)
-   real(r8) :: ndropcol(pcols)               ! column droplet number (#/m2)
-   real(r8) :: cldo_tmp, cldn_tmp
-   real(r8) :: tau_cld_regenerate
-   real(r8) :: taumix_internal_pver_inv ! 1/(internal mixing time scale for k=pver) (1/s)
-
-
-   real(r8), allocatable :: nact(:,:)  ! fractional aero. number  activation rate (/s)
-   real(r8), allocatable :: mact(:,:)  ! fractional aero. mass    activation rate (/s)
-
-   real(r8), allocatable :: raercol(:,:,:)    ! single column of aerosol mass, number mixing ratios
-   real(r8), allocatable :: raercol_cw(:,:,:) ! same as raercol but for cloud-borne phase
-
-
-   real(r8) :: na(pcols), va(pcols), hy(pcols)
-   real(r8), allocatable :: naermod(:)  ! (1/m3)
-   real(r8), allocatable :: hygro(:)    ! hygroscopicity of aerosol mode
-   real(r8), allocatable :: vaerosol(:) ! interstit+activated aerosol volume conc (cm3/cm3)
-
+   real(r8), parameter :: zkmin = 0.01_r8, zkmax = 100._r8  ! min, max vertical diffusivity [m^2/s]
+   real(r8), parameter :: wmixmin = 0.1_r8        ! minimum turbulence vertical velocity [m/s]
+
+   real(r8) :: arg        ! argument for erf test [dimensionless]
+   real(r8) :: dtinv      ! inverse time step for microphysics [s^-1]
+   real(r8) :: dtmin      ! time step to determine subloop time step [s]
+   real(r8) :: tinv       ! inverse timescale of droplet diffusivity [s^-1]
+   real(r8) :: dtt        ! timescale of droplet diffusivity [s]
+   real(r8) :: wmax       ! vertical velocity upper bound [m/s]
+   real(r8) :: dum, dumc        ! temporary variables
+   real(r8) :: tmpa             !  temporary aerosol tendency variable [s^-1]
+   real(r8) :: dact             ! cloud-borne aerosol tendency due to cloud frac tendency [# or kg/kg]
+   real(r8) :: fluxntot         ! flux of activated aerosol number into cloud [#/m^2/s]
+   real(r8) :: dtmix    ! timescale for subloop [s] 
+   real(r8) :: raertend(pver)  ! tendency of interstitial aerosol mass, number mixing ratios [# or kg/kg/s]
+   real(r8) :: qqcwtend(pver)  ! tendency of cloudborne aerosol mass, number mixing ratios [# or kg/kg/s]
```

Comment:
This unit confuse me at first glance. Maybe change to [#/kg/s or kg/kg/s]

Reply:
It might be even simpler if we separate the tendencies for mass mixing ratios from those for number mixing ratios. I don't know how tricky that would be, but it would make things easier for porting.

Reply:
That is a good point -- although I might save that for the next step of refactoring this subroutine.  The tendency to treat number mixing ratios and species mass mixing ratios in the same array structure seems to be pervasive throughout the code, though here most such fields are local variables.